### ISSUE TITLE
PHP v7.2 no longer supports mcrypt.  Unable to upgrade Monica.

### ISSUE THREAD

**Issue by virtadpt** [monicahq/monica#780]
I'm trying to upgrade Monica on my local system from a Git checkout but am unable to do so because PHP v7.2 [no longer supports mcrypt](https://wiki.php.net/rfc/mcrypt-viking-funeral) (it was flagged as deprecated in PHP v7.1.x).  When running `composer install --no-interaction --prefer-dist --optimize-autoloader` to update Monica I get the following error:

```
PHP Warning:  PHP Startup: Unable to load dynamic library 'mcrypt.so' (tried: /usr/lib/php/modules/mcrypt.so (/usr/lib/php/modules/mcrypt.so: cannot open shared object file: No such file or directory), /usr/lib/php/modules/mcrypt.so.so (/usr/lib/php/modules/mcrypt.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Your requirements could not be resolved to an installable set of packages.
```
php-mcrypt has been replaced with [php-sodium](https://wiki.php.net/rfc/libsodium), accessible through PECL.

**Comment by asbiin** [comment]
It's weird because we use Homestead/vagrant which use php 7.2, without any problem.
Are you sure your php.ini does not refer to mcrypt ?

**Comment by virtadpt** [comment]
Yes, I'm sure.  I've both visually inspected and grepped /etc/php/php.ini, and mcrypt.so is commented out.

**Comment by djaiss** [comment]
The version on https://app.monicahq.com also uses 7.2.

**Comment by virtadpt** [comment]
Due to the fact that I can't install Composer at my hosting provider, I build Monica on my laptop, compress it, upload it, and uncompress it there.

```
php 7.2.1-1
php-gd 7.2.1-1
php-intl 7.2.1-1
php-sodium 7.2.1-1
```
At my hosting provider (Dreamhost): `Fully Hosted with PHP 7.0 (User: redacted)`

However, I just updated my local repository (commit 4dfa29f8a24ce13fa281adad95212249f2761ea3) after seeing #785 and gave it another try.  I got a different error message (iconv.so isn't enabled in my /etc/php/php.ini file); I seem to have been successful at getting that commit to build.  I think it worked.  I'm going to try upgrading to see what happens.
